tests/integration/test_terraform.py

```diff
@@ -78,28 +78,88 @@ def _run_terragrunt(
     return subprocess.run(command, check=check, cwd=TERRAFORM_MODULE_DIR, env=env)
 
 
-def _wait_for_microceph_units(juju: jubilant.Juju, *, expected_units: int) -> None:
-    def _has_units(status: jubilant.Status) -> bool:
-        app = status.apps.get(APP_NAME)
-        if not app or not app.units:
-            return False
-        return len(app.units) >= expected_units
-
-    juju.wait(_has_units, error=jubilant.any_error, timeout=DEFAULT_TIMEOUT)
+def _microceph_services_snapshot(
+    juju: jubilant.Juju, unit_name: str
+) -> tuple[list[set[str]], str]:
+    output = juju.ssh(unit_name, "sudo", "microceph", "status")
+    services: list[set[str]] = []
+    for line in output.splitlines():
+        stripped = line.strip()
+        if stripped.lower().startswith("services:"):
+            _, entries = stripped.split(":", 1)
+            names = [svc.strip().lower() for svc in entries.split(",") if svc.strip()]
+            services.append(set(names))
+    return services, output
+
+
+def _status_reports_rgw(status: dict[str, Any]) -> bool:
```

Comment:
can this function be renamed to something like `is_rgw_active` or something simpler ?

Reply:
I feel like is_rgw_active isn't quite accurate b/c we don't know if it is, but I get your point, maybe is_rgw_enabled()?